code/modules/asset_cache/asset_list_items.dm

```diff
@@ -445,6 +451,10 @@
 			stack_trace("[item] does not have a valid icon state, icon=[icon_file], icon_state=[json_encode(icon_state)](\ref[icon_state]), icon_states=[icon_states_string]")
 			I = icon('icons/turf/floors.dmi', "", SOUTH)
 
+		if(is_gags_icon)
+			// Scales the icon to 32x32, as it is 0x0 by default. Type checks can be added here if other sizes are necessary.
+			I.Scale(32, 32)
```

Comment:
GAGS supports sizes other than 32x32, was converting the icon to an image not a good way of getting width and height?

Reply:
Haven't really got that approach to work particularly well to be honest. I think considering vendor icons are typically 32x32 this is a sensible solution, as from what I know none of the vendors use a different size.

Reply:
This approach also allows avoiding a potentially more expensive operation of converting the icon to an image that ends up not getting used later on, as the sprite sheet function accepts only icons, not images